### ISSUE TITLE
onIosBackground not triggered

### ISSUE THREAD

**Issue by kenthken** [ekasetiawans/flutter_background_service#437]
I face an issue where for iOS cant execute the task inside the onIosBackground function.

![Screenshot 2024-05-24 155136](https://github.com/ekasetiawans/flutter_background_service/assets/97033788/af16ffe6-de5b-470b-9da9-fd03d789dd26)

![Screenshot 2024-05-24 155229](https://github.com/ekasetiawans/flutter_background_service/assets/97033788/3094c9c2-a058-4325-a93c-9805554a5120)

![Screenshot 2024-05-24 155334](https://github.com/ekasetiawans/flutter_background_service/assets/97033788/d4ff420d-2ae3-4b70-9499-e573f0cdabba)

![Screenshot 2024-05-24 155422](https://github.com/ekasetiawans/flutter_background_service/assets/97033788/27c1fce7-425e-45dc-b657-1a247841e803)

**Comment by kenthken** [comment]
even after > 15 minutes. any idea how to solve it?

**Comment by alex-lechner** [comment]
I have the same problem but I haven't found a solution yet.

**Comment by kenthken** [comment]
> I have the same problem but I haven't found a solution yet.

I think there is no solution about this case, since it was design that way by the iOS.
Maybe you can watch the video documentation by Apple Developer

https://developer.apple.com/videos/play/wwdc2019/707/

you can skip it until background fetch section above video.

**Comment by alex-lechner** [comment]
Ohhh wow, thank you! It's along video so I will be watching it another time.

I think this package is not actively maintained anymore, right @ekasetiawans?

There are many issues open and the last update of the package was a while ago. While it's still working like a charm, there are bugs like the `onIosBackground` which do not seem to be fixed anytime soon.

I have found some possible alternatives where the background execution of tasks on iOS might work:

- https://github.com/fluttercommunity/flutter_workmanager?tab=readme-ov-file
- https://github.com/transistorsoft/flutter_background_fetch
- Tutorial for native implementation: https://medium.com/flutter/executing-dart-in-the-background-with-flutter-plugins-and-geofencing-2b3e40a1a124

**Comment by alex-lechner** [comment]
> @alex-lechner i wish i have enough time to maintenance

@ekasetiawans Do you have any plans on updating it or handing it over to one of the contributors so it can be maintained?

**Comment by kenthken** [comment]
> Ohhh wow, thank you! It's along video so I will be watching it another time.
> 
> I think this package is not actively maintained anymore, right @ekasetiawans?
> 
> There are many issues open and the last update of the package was a while ago. While it's still working like a charm, there are bugs like the `onIosBackground` which do not seem to be fixed anytime soon.
> 
> I have found some possible alternatives where the background execution of tasks on iOS might work:
> 
> * https://github.com/fluttercommunity/flutter_workmanager?tab=readme-ov-file
> * https://github.com/transistorsoft/flutter_background_fetch
> * Tutorial for native implementation: https://medium.com/flutter/executing-dart-in-the-background-with-flutter-plugins-and-geofencing-2b3e40a1a124

have u try another package that might work?

**Comment by alex-lechner** [comment]
I haven't tried another package yet and I also have read that iOS will execute the background fetch at a minimum every 15 minutes but - as you have already mentioned - based on the user's app usage. However, you have mentioned above that even after 15 minutes the background fetch does not get executed.

**Comment by kenthken** [comment]
Yes correct, minimum is 15 minutes, but so far that only happened once in my case, after that it will take like 2 hours or even the next day for the iOS to execute again.

After a few research, when you just initiate the iOS on background fetch, it will take minimum 1 day for the OS to read and analyst the apps behavior by the users and finally the execution will be start in the next day, and when the issue that I created at the first place  “after 15 minutes not executed” is when the OS not finish to analyst my app, and the execution is happened in the next day of this issues is created. @alex-lechner
